### PR TITLE
fix: VSCode Jest debugging

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Configure `settings.json` and `launch.json` to bypass `tools/jest.mjs` wrapper
- Provide reasonable options/env
- Fix running/debugging particular tests from the test explorer or side gutter

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: .vscode/launch.json [hunk 1]
```diff
@@ -84,14 +84,29 @@
...
       "console": "integratedTerminal",
-      "internalConsoleOptions": "neverOpen",
+      "internalConsoleOptions": "openOnSessionStart",
       "cwd": "${workspaceFolder}",
-      "runtimeExecutable": "pnpm",
-      "program": "jest:vscode",
-      "args": ["--runInBand", "--watchAll=false", "--testTimeout=100000000"]
+      "runtimeExecutable": "node",
+      "runtimeArgs": ["--experimental-vm-modules"],
+      "program": "node_modules/jest/bin/jest.js",
```
Comment (viceice) [inline .vscode/launch.json:94]: why not simply jest, like the others?

**Review by viceice — COMMENTED** [review]
check the jest.mjs, you need an additional node arg to get the RSA decrypt working

File: .vscode/launch.json (hunk 1 above)
Comment (zharinov) [reply .vscode/launch.json:94]: I think the process spawn caused some weird effects for how breakpoints work, while the direct call makes debug test runs more straightforward (and it feels faster)

Comment (zharinov) [reply .vscode/launch.json:94]: Do we really need that CVE node argument for test debugging? It produces annoying message and looks unnecessary for majority of tests that someone could run from the VSCode UI.

File: .vscode/settings.json [hunk 2]
```diff
@@ -16,8 +16,13 @@
     ".releaserc": "json"
   },
   "omnisharp.autoStart": false,
-  "jest.autoRun": "off",
-  "jest.jestCommandLine": "pnpm jest",
+  "jest.runMode": "on-demand",
+  "jest.jestCommandLine": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
```
Comment (viceice) [inline .vscode/settings.json:20]: doesn't this cause import issues with pnpm?

**Review by viceice — COMMENTED** [review]
🤔 last time I got import issues when running jest without pnpm because of the missing node module env

File: .vscode/settings.json (hunk 2 above)
Comment (zharinov) [reply .vscode/settings.json:20]: Just managed to run the entire test set from the UI with no problems 🤷‍♂️

**Review by viceice — APPROVED** [review]
🤷‍♂️

**Pull request merged**